### PR TITLE
Add a conference loading layer

### LINKED ISSUE #135: Voxxed Days Ticino schedule is shown for Devoxx Belgium
1) Look at the Voxxed Days Ticino schedule 
2) Switch conference and select Devoxx Belgium
3) You'll notice that the previous schedule is still shown

image

image

### PR DESCRIPTION
**Pull request by abhinayagarwal**
This PR introduces a conference loading layer which is shown when a conference is selected by the user. The layer will hide itself either when loading of sessions is completed or timeout of 20 seconds has reached.

Fixes #135

### CODE REVIEW

File: DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    private ConferenceLoadingLayer(Service service, String conferenceName) {
+
+        this.glassPane = MobileApplication.getInstance().getGlassPane();
+        progressIndicator = new ProgressIndicator();
+
+        conferenceLabel = new Label(conferenceName);
+        conferenceLabel.setWrapText(true);
+        conferenceLabel.setTextAlignment(TextAlignment.CENTER);
+        conferenceLabel.setFont(Font.font(Font.getDefault().getFamily(), FontWeight.BOLD, 15));
```
Comment (jperedadnr) [inline DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java]: Font is too small on tablet.
Maybe this should be moved (later on?) to CSS.

File: DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    private final ProgressIndicator progressIndicator;
+    private final Label conferenceLabel;
+    private Region background;
+    private final DisplayService displayService;
+
+    private ConferenceLoadingLayer(Service service, String conferenceName) {
+
+        this.glassPane = MobileApplication.getInstance().getGlassPane();
+        progressIndicator = new ProgressIndicator();
+
```
[the remark this answers is not in the document]
Comment (abhinayagarwal) [reply DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java]: Fixed

File: DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java (hunk 1 above)
Comment (abhinayagarwal) [reply DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java]: I have set it to 1.3em on `common_tablet.css`

File: DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                    service.retrieveSessions().removeListener(this);
+                }
+            }
+        };
+        timeout.setOnFinished(e -> {
+            service.retrieveSessions().removeListener(sessionsListener);
+            hide();
+        });
+        timeout.playFromStart();
+        service.retrieveSessions().addListener(sessionsListener);
```
Comment (jperedadnr) [inline DevoxxClientMobile/src/main/java/com/devoxx/views/layer/ConferenceLoadingLayer.java]: Can you add a listener to the RF failure and stop the transition after that? I see that the RF fails early (for a future conference with no results), but you still wait the whole TIMEOUT.

**Comment by jperedadnr** [comment]
Works fine now, probably the layer is closed too fast. This could be improved later on adding a short delay or some transitions.

**Comment by abhinayagarwal** [comment]
> Works fine now, probably the layer is closed too fast. This could be improved later on adding a short delay or some transitions.

@jperedadnr I am calling `super.hide()` at the end inside `hide()`  for this specific reason. On my mobile the layer is closed once all the items are loaded in the CLV.

Is there a specific case where it hides early?

**Comment by jperedadnr** [comment]
I didn't say early, but when the content is loaded too fast, the layer shows up and the progress indicator is barely visible for less than a second. Then, it is hidden and the catalog is shown.
I was suggesting to create a more smooth transition between layer and view, keeping the progress indicator at least for one full cycle won't hurt, and then fade in the layer out. But of course, this can be discussed in a follow-up issue.

**Pull request merged**